### PR TITLE
Increasing type coverage for FTP provider

### PR DESCRIPTION
**Pull request by mlgruby**
This PR is about increasing typing coverage for the FTP provider. Part of: #9708 

CC: @kaxil @turbaszek @mik-laj Can I ask for a review of this PR? 

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

### CODE REVIEW

**Comment by mlgruby** [comment]
Below is the type coverage for FTP provider: 

0.000000 dingding
0.000000 opsgenie
0.000000 presto
0.000000 qubole
0.000000 samba
0.000000 zendesk
16.666650 ssh
16.666667 oracle
16.666667 yandex
16.666700 odbc
21.428550 databricks
25.000000 exasol
25.000000 singularity
25.000000 snowflake
27.380950 slack
33.333333 sftp
41.074865 amazon
41.666650 plexus
46.153850 imap
52.078754 microsoft
67.298000 cncf
67.857150 salesforce
69.100860 google
75.000000 jdbc
75.000000 postgres
80.625000 ftp  =================> from 71 to 80
83.333300 openfaas
87.421875 apache
87.500000 redis
88.333340 mysql
88.888900 papermill
90.909100 elasticsearch
96.875000 mongo
100.000000 celery
100.000000 cloudant
100.000000 datadog
100.000000 discord
100.000000 docker
100.000000 facebook
100.000000 grpc
100.000000 hashicorp
100.000000 http
100.000000 jenkins
100.000000 jira
100.000000 pagerduty
100.000000 segment
100.000000 sendgrid
100.000000 sqlite
100.000000 vertica

File: airflow/providers/ftp/hooks/ftp.py
Comment (kaxil) [inline airflow/providers/ftp/hooks/ftp.py]: Just returns int as it is the size

```suggestion
    def get_size(self, path: str) -> int:
```

**Review by kaxil — COMMENTED** [review]
great to see you here Satya :)

File: airflow/providers/ftp/hooks/ftp.py
Comment (mlgruby) [reply airflow/providers/ftp/hooks/ftp.py]: because of implicit return of string from size() I have put an option of either int or str.

Comment (kaxil) [reply airflow/providers/ftp/hooks/ftp.py]: https://docs.python.org/3/library/ftplib.html#ftplib.FTP.size

It will size in bytes or None, so 

```suggestion
    def get_size(self, path: str) -> Optional[int]:
```

It won't return string

Comment (mlgruby) [reply airflow/providers/ftp/hooks/ftp.py]: True, but mypy throws an error for `int` or `Optional[int]` return type as there is an implicit string print statement in this function. I can change it to `Any` (but that will again a dynamic typed one)

Comment (kaxil) [reply airflow/providers/ftp/hooks/ftp.py]: Mypy is going crazy, I would suggest keeping it as 

```
def get_size(self, path: str) -> Optional[int]:
```

and do a cast on `return conn.size(path)` as below on L259:

```
return int(conn.size(path))
```

Comment (mlgruby) [reply airflow/providers/ftp/hooks/ftp.py]: Your suggestion needs explicit casting of return statement to `int` 
```suggestion
    def get_size(self, path: str) -> Optional[int]:
        conn = self.get_conn()
        return int(conn.size(path) or 0)
```

Comment (mlgruby) [reply airflow/providers/ftp/hooks/ftp.py]: Well, I would suggest to not do type checking for this function as we have more than 75% coverage for FTP provider. What's your thought?

Comment (kaxil) [reply airflow/providers/ftp/hooks/ftp.py]: or 
```suggestion
    def get_size(self, path: str) -> Optional[int]:
        conn = self.get_conn()
        size = conn.size(path)
        return int(size) if size else None
```

Comment (mlgruby) [reply airflow/providers/ftp/hooks/ftp.py]: that works 👍🏼 Will make the changes!

Comment (kaxil) [reply airflow/providers/ftp/hooks/ftp.py]: The above suggestion should work, just tested it with mypy. 
```
Run mypy.................................................................Passed
```

Once that is done, I will merge this PR :)

File: airflow/providers/ftp/hooks/ftp.py
Comment (kaxil) [inline airflow/providers/ftp/hooks/ftp.py]:
```suggestion
from typing import List, Optional, Any
```

we can remove Union once https://github.com/apache/airflow/pull/11107/files#r493825856 is implemented

File: airflow/providers/ftp/hooks/ftp.py
Comment (mlgruby) [reply airflow/providers/ftp/hooks/ftp.py]: Thanks brother, let me start my journey into Airflow from here :)

**Review by kaxil — APPROVED** [review]
Awesome 🚀

**Comment by kaxil** [comment]
Just waiting for the CI before I merge.

Looking forward for more PRs :)

**Review by turbaszek — APPROVED** [review]

**Pull request merged**
